### PR TITLE
gpy: init at 1.7.7

### PR DESCRIPTION
**Pull request by bcdarwin**
###### Things done

Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers.

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [N/A] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [N/A] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [N/A] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/python-modules/gpy/default.nix [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  pname = "GPy";
+  version = "1.7.7";
+  name  = "${pname}-${version}";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1b4siirlkqic1lsn9bi9mnp8fpbpw1ijwv0z2i6r2zdrk3d6szs1";
+  };
+
+  doCheck = false;
```
Comment (FRidh) [inline pkgs/development/python-modules/gpy/default.nix]: Why are the tests disabled? Always include a comment explaining why.

Comment (bcdarwin) [inline pkgs/development/python-modules/gpy/default.nix]: I've fixed the tests.

Comment (bcdarwin) [inline pkgs/development/python-modules/gpy/default.nix]: Never mind, something is still wrong.

**Comment by bcdarwin** [comment]
I've disabled the tests for now and added a comment regarding this.

File: pkgs/development/python-modules/gpy/default.nix [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+  buildInputs = [ nose cython matplotlib ];
+
+  propagatedBuildInputs = [ numpy scipy six paramz ];
+
+  meta = with stdenv.lib; {
+    description = "Gaussian process framework in Python";
+    homepage = https://sheffieldml.github.io/GPy;
+    license = licenses.bsd3;
+    maintainers = with maintainers; [ bcdarwin ];
```
Comment (matthewbauer) [inline pkgs/development/python-modules/gpy/default.nix]: We also need a platforms attribute here.

Comment (FRidh) [inline pkgs/development/python-modules/gpy/default.nix]: Why? Do you want to exclude a certain platform? `buildPythonPackage` inherits `platforms` from `python`.

File: pkgs/development/python-modules/gpy/default.nix [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    inherit pname version;
+    sha256 = "1b4siirlkqic1lsn9bi9mnp8fpbpw1ijwv0z2i6r2zdrk3d6szs1";
+  };
+
+  # running tests produces "ImportError: cannot import name 'linalg_cython'"
+  # even though Cython has run
+  checkPhase = "nosetests -d";
+  doCheck = false;
+
+  buildInputs = [ nose cython matplotlib ];
```
Comment (FRidh) [inline pkgs/development/python-modules/gpy/default.nix]: could you move `nose` to `checkInputs`

File: pkgs/development/python-modules/gpy/default.nix (hunk 2 above)
Comment (matthewbauer) [inline pkgs/development/python-modules/gpy/default.nix]: Oh nevermind, didn't realize it was python.

File: pkgs/development/python-modules/gpy/default.nix (hunk 3 above)
Comment (bcdarwin) [inline pkgs/development/python-modules/gpy/default.nix]: Done; I've also moved `matplotlib` to `propagatedBuildInputs` as it's both required for tests and by some `gpy` modules.

**Pull request merged**